### PR TITLE
Add tags to environment

### LINKED ISSUE #190: CloudFormation: Add tags for environments
In CreateChangeSet, we should provide tags. This will let us transitively tag all our created resources. 

The tags for environment should be:

`ecs-project:{project name}`, `ecs-env:{env-name}`

[CF Documentation Ref](https://docs.aws.amazon.com/sdk-for-go/api/service/cloudformation/#CreateChangeSetInput)

### PR DESCRIPTION
**Pull request by SoManyHs**
Fixes #190 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by SoManyHs** [comment]
Ran:
```
~/workspace/amazon-ecs-archer(tag-the-environment)$ archer init
Note: It's best to run this command in the root of your workspace.
Welcome the the ECS CLI! We're going to walk you through some questions to help you get set up
with a project on ECS. A project is a collection of containerized applications (or micro-services) 
that operate together.

Looks like you don't have any existing projects. Let's create one!
? What would you like to call your project? BadGoose
? What type of application do you want to make?  [Use arrows to move, type to filter, ? for more ? What type of application do you want to make? Load Balanced Web App
? What do you want to call this Load Balanced Web App? Honker
Ok great, we'll set up a Load Balanced Web App named Honker in project BadGoose.

✔ Success! Wrote the manifest for Honker app at 'ecs-project/Honker-app.yml'

All right, you're all set for local development.
? Would you like to deploy a staging environment? Yes
Done!
⠹ Deploying env... 2019/10/10 23:15:45 Created Environment with version 1
Done!

```

Verified that the new tags were added to the changeset:

```
aws cloudformation describe-change-set --change-set-name arn:aws:cloudformation:us-west-2:xxxxxxxx151:changeSet/ecscli-dd988eb7-948c-41e2-a51f-3eeb78ad6b59/29656dec-7fcd-4df5-aa50-6b5f1edca9ed

{
    "StackId": "arn:aws:cloudformation:us-west-2:xxxxxxxxx151:stack/BadGoose-test/ff0190e0-ebed-11e9-9aa8-0ad20430b590", 
    "Status": "CREATE_COMPLETE", 
    "ChangeSetName": "ecscli-dd988eb7-948c-41e2-a51f-3eeb78ad6b59", 
    "Description": null, 
    "Parameters": [
        {
            "ParameterValue": "10.0.2.0/24", 
            "ParameterKey": "PrivateSubnet1CIDR"
        }, 
        {
            "ParameterValue": "true", 
            "ParameterKey": "IncludePublicLoadBalancer"
        }, 
        {
            "ParameterValue": "10.0.1.0/24", 
            "ParameterKey": "PublicSubnet2CIDR"
        }, 
        {
            "ParameterValue": "10.0.0.0/16", 
            "ParameterKey": "VpcCIDR"
        }, 
        {
            "ParameterValue": "10.0.3.0/24", 
            "ParameterKey": "PrivateSubnet2CIDR"
        }, 
        {
            "ParameterValue": "10.0.0.0/24", 
            "ParameterKey": "PublicSubnet1CIDR"
        }
    ], 
    "Tags": [
        {
            "Value": "BadGoose", 
            "Key": "ecs-project"
        }, 
        {
            "Value": "test", 
            "Key": "ecs-env"
        }
    ], 
    "ChangeSetId": "arn:aws:cloudformation:us-west-2:794715269151:changeSet/ecscli-dd988eb7-948c-41e2-a51f-3eeb78ad6b59/29656dec-7fcd-4df5-aa50-6b5f1edca9ed", 
    "CreationTime": "2019-10-11T06:11:43.236Z", 
    "StatusReason": null, 
    "StackName": "BadGoose-test", 
    "NotificationARNs": [], 
    "Capabilities": [
        "CAPABILITY_IAM"
    ], 
    "ExecutionStatus": "EXECUTE_COMPLETE", 
    "Changes": [
        {
            "ResourceChange": {
                "Action": "Add", 
                "ResourceType": "AWS::ECS::Cluster", 
                "Scope": [], 
                "Details": [], 
                "LogicalResourceId": "Cluster"
            }, 
            "Type": "Resource"
        }, 
        {
            "ResourceChange": {
                "Action": "Add", 
                "ResourceType": "AWS::EC2::Route", 
                "Scope": [], 
                "Details": [], 
                "LogicalResourceId": "DefaultPrivateRoute1"
            }, 
            "Type": "Resource"
        }, 
        {
            "ResourceChange": {
                "Action": "Add", 
                "ResourceType": "AWS::EC2::Route", 
                "Scope": [], 
                "Details": [], 
                "LogicalResourceId": "DefaultPrivateRoute2"
            }, 
            "Type": "Resource"
        }, 
        {
            "ResourceChange": {
                "Action": "Add", 
                "ResourceType": "AWS::EC2::Route", 
                "Scope": [], 
                "Details": [], 
                "LogicalResourceId": "DefaultPublicRoute"
            }, 
            "Type": "Resource"
        }, 
        {
            "ResourceChange": {
                "Action": "Add", 
                "ResourceType": "AWS::EC2::InternetGateway", 
                "Scope": [], 
                "Details": [], 
                "LogicalResourceId": "InternetGateway"
            }, 
            "Type": "Resource"
        }, 
        {
            "ResourceChange": {
                "Action": "Add", 
                "ResourceType": "AWS::EC2::VPCGatewayAttachment", 
                "Scope": [], 
                "Details": [], 
                "LogicalResourceId": "InternetGatewayAttachment"
            }, 
            "Type": "Resource"
        }, 
        {
            "ResourceChange": {
                "Action": "Add", 
                "ResourceType": "AWS::EC2::NatGateway", 
                "Scope": [], 
                "Details": [], 
                "LogicalResourceId": "NATGateway1"
            }, 
            "Type": "Resource"
        }, 
        {
            "ResourceChange": {
                "Action": "Add", 
                "ResourceType": "AWS::EC2::NatGateway", 
                "Scope": [], 
                "Details": [], 
                "LogicalResourceId": "NATGateway2"
            }, 
            "Type": "Resource"
        }, 
        {
            "ResourceChange": {
                "Action": "Add", 
                "ResourceType": "AWS::EC2::EIP", 
                "Scope": [], 
                "Details": [], 
                "LogicalResourceId": "NatGateway1EIP"
            }, 
            "Type": "Resource"
        }, 
        {
            "ResourceChange": {
                "Action": "Add", 
                "ResourceType": "AWS::EC2::EIP", 
                "Scope": [], 
                "Details": [], 
                "LogicalResourceId": "NatGateway2EIP"
            }, 
            "Type": "Resource"
        }, 
        {
            "ResourceChange": {
                "Action": "Add", 
                "ResourceType": "AWS::EC2::RouteTable", 
                "Scope": [], 
                "Details": [], 
                "LogicalResourceId": "PrivateRouteTable1"
            }, 
            "Type": "Resource"
        }, 
        {
            "ResourceChange": {
                "Action": "Add", 
                "ResourceType": "AWS::EC2::RouteTable", 
                "Scope": [], 
                "Details": [], 
                "LogicalResourceId": "PrivateRouteTable2"
            }, 
            "Type": "Resource"
        }, 
        {
            "ResourceChange": {
                "Action": "Add", 
                "ResourceType": "AWS::EC2::Subnet", 
                "Scope": [], 
                "Details": [], 
                "LogicalResourceId": "PrivateSubnet1"
            }, 
            "Type": "Resource"
        }, 
        {
            "ResourceChange": {
                "Action": "Add", 
                "ResourceType": "AWS::EC2::SubnetRouteTableAssociation", 
                "Scope": [], 
                "Details": [], 
                "LogicalResourceId": "PrivateSubnet1RouteTableAssociation"
            }, 
            "Type": "Resource"
        }, 
        {
            "ResourceChange": {
                "Action": "Add", 
                "ResourceType": "AWS::EC2::Subnet", 
                "Scope": [], 
                "Details": [], 
                "LogicalResourceId": "PrivateSubnet2"
            }, 
            "Type": "Resource"
        }, 
        {
            "ResourceChange": {
                "Action": "Add", 
                "ResourceType": "AWS::EC2::SubnetRouteTableAssociation", 
                "Scope": [], 
                "Details": [], 
                "LogicalResourceId": "PrivateSubnet2RouteTableAssociation"
            }, 
            "Type": "Resource"
        }, 
        {
            "ResourceChange": {
                "Action": "Add", 
                "ResourceType": "AWS::ElasticLoadBalancingV2::LoadBalancer", 
                "Scope": [], 
                "Details": [], 
                "LogicalResourceId": "PublicLoadBalancer"
            }, 
            "Type": "Resource"
        }, 
        {
            "ResourceChange": {
                "Action": "Add", 
                "ResourceType": "AWS::EC2::SecurityGroup", 
                "Scope": [], 
                "Details": [], 
                "LogicalResourceId": "PublicLoadBalancerSecurityGroup"
            }, 
            "Type": "Resource"
        }, 
        {
            "ResourceChange": {
                "Action": "Add", 
                "ResourceType": "AWS::EC2::RouteTable", 
                "Scope": [], 
                "Details": [], 
                "LogicalResourceId": "PublicRouteTable"
            }, 
            "Type": "Resource"
        }, 
        {
            "ResourceChange": {
                "Action": "Add", 
                "ResourceType": "AWS::EC2::Subnet", 
                "Scope": [], 
                "Details": [], 
                "LogicalResourceId": "PublicSubnet1"
            }, 
            "Type": "Resource"
        }, 
        {
            "ResourceChange": {
                "Action": "Add", 
                "ResourceType": "AWS::EC2::SubnetRouteTableAssociation", 
                "Scope": [], 
                "Details": [], 
                "LogicalResourceId": "PublicSubnet1RouteTableAssociation"
            }, 
            "Type": "Resource"
        }, 
        {
            "ResourceChange": {
                "Action": "Add", 
                "ResourceType": "AWS::EC2::Subnet", 
                "Scope": [], 
                "Details": [], 
                "LogicalResourceId": "PublicSubnet2"
            }, 
            "Type": "Resource"
        }, 
        {
            "ResourceChange": {
                "Action": "Add", 
                "ResourceType": "AWS::EC2::SubnetRouteTableAssociation", 
                "Scope": [], 
                "Details": [], 
                "LogicalResourceId": "PublicSubnet2RouteTableAssociation"
            }, 
            "Type": "Resource"
        }, 
        {
            "ResourceChange": {
                "Action": "Add", 
                "ResourceType": "AWS::EC2::VPC", 
                "Scope": [], 
                "Details": [], 
                "LogicalResourceId": "VPC"
            }, 
            "Type": "Resource"
        }
    ], 
    "RollbackConfiguration": {}
}
```

File: internal/pkg/deploy/cloudformation/changeset.go [hunk 1]
```diff
@@ -101,7 +102,7 @@ func (set *changeSet) execute() error {
 // createChangeSetOpt is a functional option to add additional settings to a CreateChangeSetInput.
 type createChangeSetOpt func(in *cloudformation.CreateChangeSetInput)
 
-func createChangeSetInput(stackName, templateBody string, options ...createChangeSetOpt) (*cloudformation.CreateChangeSetInput, error) {
+func createChangeSetInput(env *archer.Environment, templateBody string, options ...createChangeSetOpt) (*cloudformation.CreateChangeSetInput, error) {
```
Comment (efekarakus) [inline internal/pkg/deploy/cloudformation/changeset.go]: Can we keep this as stackName? The intent of `createChangeSetInput` is to be independent of an `environment` or `application` (I should have added comments :P)

Can we instead create a `createChangeSetOpt` that applies tags?
```go
// in changeset.go
func withEnvTags(env *archer.Environment) createChangeSetOpt {
	return func(in *cloudformation.CreateChangeSetInput) {
		in.tags = []*cloudformation.Tag{
                        {
			         Key:   aws.String("ecs-project"),
			         Value: aws.String(env.Project),
		        },
		       {
			         Key:   aws.String("ecs-env"),
			         Value: aws.String(env.Name),
		        },
	       }
	}
}
```
Then `DeployEnvironment` can use this function at l.54

**Comment by SoManyHs** [comment]
```
~/ecs/ecs-cli/testing/archer1$ archer env add
2019/10/14 16:59:33 couldn't find a directory called ecs-project up to 5 levels up from /Users/hsinghui/ecs/ecs-cli/testing/archer1
? What is your project's name? BadGoose
? What is your environment's name? testing
Done!
⠸ Deploying env... 2019/10/14 17:04:12 Created Environment with version 1
Done!

~/ecs/ecs-cli/testing/archer1$ aws cloudformation describe-change-set --change-set-name arn:aws:cloudformation:us-west-2:xxxxxxxx51:changeSet/ecscli-8e8a21be-4233-4126-bfa4-087c12b14865/736b6712-c36c-4da8-9b23-f05048c33a1b
{
    "StackId": "arn:aws:cloudformation:us-west-2:xxxxxxxxxx51:stack/BadGoose-testing/af809220-eede-11e9-9aa8-0ad20430b590", 
    "Status": "CREATE_COMPLETE", 
    "ChangeSetName": "ecscli-8e8a21be-4233-4126-bfa4-087c12b14865", 
    "Description": null, 
    "Parameters": [
        {
            "ParameterValue": "10.0.2.0/24", 
            "ParameterKey": "PrivateSubnet1CIDR"
        }, 
        {
            "ParameterValue": "true", 
            "ParameterKey": "IncludePublicLoadBalancer"
        }, 
        {
            "ParameterValue": "10.0.1.0/24", 
            "ParameterKey": "PublicSubnet2CIDR"
        }, 
        {
            "ParameterValue": "10.0.0.0/16", 
            "ParameterKey": "VpcCIDR"
        }, 
        {
            "ParameterValue": "10.0.3.0/24", 
            "ParameterKey": "PrivateSubnet2CIDR"
        }, 
        {
            "ParameterValue": "10.0.0.0/24", 
            "ParameterKey": "PublicSubnet1CIDR"
        }
    ], 
    "Tags": [
        {
            "Value": "BadGoose", 
            "Key": "ecs-project"
        }, 
        {
            "Value": "testing", 
            "Key": "ecs-env"
        }
    ], 
    "ChangeSetId": "arn:aws:cloudformation:us-west-2:xxxxxxxxx51:changeSet/ecscli-8e8a21be-4233-4126-bfa4-087c12b14865/736b6712-c36c-4da8-9b23-f05048c33a1b", 
    "CreationTime": "2019-10-14T23:59:40.864Z", 
    "StatusReason": null, 
    "StackName": "BadGoose-testing", 
```

File: internal/pkg/deploy/cloudformation/changeset.go [hunk 2]
```diff
@@ -110,8 +111,9 @@ func createChangeSetInput(stackName, templateBody string, options ...createChang
 	// The change set name must match the regex [a-zA-Z][-a-zA-Z0-9]*. The generated UUID can start with a number,
 	// by prefixing the uuid with a word we guarantee that we start with a letter.
 	name := fmt.Sprintf("%s-%s", "ecscli", id.String())
+
 	in := &cloudformation.CreateChangeSetInput{
-		Capabilities:  []*string{aws.String(cloudformation.CapabilityCapabilityIam)},
+		Capabilities:  aws.StringSlice([]string{cloudformation.CapabilityCapabilityIam}),
```
Comment (efekarakus) [inline internal/pkg/deploy/cloudformation/changeset.go]: :O

File: internal/pkg/deploy/cloudformation/changeset.go [hunk 3]
```diff
@@ -133,3 +135,18 @@ func withCreateChangeSetType() createChangeSetOpt {
...
 }
+
+func withEnvTags(env *archer.Environment) createChangeSetOpt {
+	return func(in *cloudformation.CreateChangeSetInput) {
+		in.Tags = []*cloudformation.Tag{
+                        {
+			         Key:   aws.String("ecs-project"),
+			         Value: aws.String(env.Project),
+		        },
+		       {
```
Comment (sonofachamp) [inline internal/pkg/deploy/cloudformation/changeset.go]: Formatting looks off here.

**Pull request merged**